controllers/dbaas.redhat.com/crunchybridgeinventory_controller.go

```diff
@@ -62,14 +66,80 @@ func (r *CrunchyBridgeInventoryReconciler) Reconcile(ctx context.Context, req ct
 		logger.Error(err, "Error fetching CrunchyBridgeInventory for reconcile")
 		return ctrl.Result{}, err
 	}
-	//
 
+	bridgeapiClient, err := r.setupClient(inventory, logger)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Crunchy Bridge Client Configured ")
+	err = r.discoverInventories(&inventory, bridgeapiClient, logger)
+	if err != nil {
+		statusErr, updated := r.updateStatus(ctx, req, inventory, metav1.ConditionFalse, BackendError, err.Error())
+		if !updated {
+			return ctrl.Result{Requeue: true}, statusErr
+		}
+		logger.Error(err, "Error while querying the inventory from CrunchyBridge")
+		return ctrl.Result{}, err
+	}
+	statusErr, updated := r.updateStatus(ctx, req, inventory, metav1.ConditionTrue, SyncOK, SuccessMessage)
+	if !updated {
+		return ctrl.Result{Requeue: true}, statusErr
+	}
 	return ctrl.Result{}, nil
 }
 
+func (r *CrunchyBridgeInventoryReconciler) setupClient(inventory dbaasredhatcomv1alpha1.CrunchyBridgeInventory, logger logr.Logger) (*bridgeapi.Client, error) {
+	baseUrl, err := url.Parse(r.APIBaseURL)
+	if err != nil {
+		logger.Error(err, "Malformed URL reconcile")
+		return nil, err
+	}
+	KubeSecretCredentialProvider := &kubeadapter.KubeSecretCredentialProvider{
+		Client:      r.Client,
+		Namespace:   inventory.Spec.CredentialsRef.Namespace,
+		Name:        inventory.Spec.CredentialsRef.Name,
+		KeyField:    "publicApiKey",
+		SecretField: "privateApiSecret",
+	}
+	KubeSecretCredentialProvider.ProvideCredential()
```

Comment:
I think this line can be deleted as nothing is done with the return value.

Reply:
Removed.